### PR TITLE
Support and examples for Class-Based Views (HTTPMethodView)

### PR DESCRIPTION
**Pull request by Guoli-Lyu**
Add support for Class-Based Views as issues #12  and #42 have pointed out.

In addition, a simple example is added to demonstrate that in class based view, sanic-openapi can actually work.

Examples for class-based views is also added to car examples.

### CODE REVIEW

**Comment by tudormunteanu** [comment]
Any idea when this will be merged in?

**Comment by notnami** [comment]
Is this adding something different than #67?

**Comment by Guoli-Lyu** [comment]
You can install forked version via pip and see what's different. @notnami 

I think that the main difference are examples and tags support for blueprint. 

And I didn't notice that there had been some PR trying to provide class-based views support when I wrote my own solution.

**Comment by Guoli-Lyu** [comment]
@tudormunteanu I think that sanic-openapi is no longer under maintenance. Besides, I need support for OpenAPI 3 for bearer token authentication. And that's why I turn to api-spec with some tricky code. 
If anyone is interested, I can provide an example to show how to incorporate api-spec into sanic as it does not provide official support for sanic.

**Comment by logileifs** [comment]
Can we get this pull request or [67](https://github.com/huge-success/sanic-openapi/pull/67) merged?
Is there something I can do to help? I could resolve the conflicts if that's the only thing keeping this from being merged

**Comment by ahopkins** [comment]
@chenjr0719 Can you take a look at these two and decide which one needs to be merged?

@logileifs The conflicts, but also understanding the overlap with #67.

File: sanic_openapi/openapi.py [hunk 1]
```diff
@@ -47,10 +48,21 @@ def build_spec(app, loop):
...
         if hasattr(blueprint, 'routes'):
             for route in blueprint.routes:
-                route_spec = route_specs[route.handler]
-                route_spec.blueprint = blueprint
-                if not route_spec.tags:
-                    route_spec.tags.append(blueprint.name)
+                if hasattr(route.handler, 'view_class'):
+                    # class based view
+                    view = route.handler.view_class
+                    for http_method in HTTP_METHODS:
```
Comment (chenjr0719) [inline sanic_openapi/openapi.py]: BTW, here is a replacement if we don't want to import HTTP_METHODS:
```python
for http_method in route.methods
```

**Comment by Guoli-Lyu** [comment]
I'm glad that this PR is receiving serious attention to bring class-based views to sanic-openapi. This PR  has been around for a long time and may not match the current code of sanic-openapi.

So what can I help?

**Comment by chenjr0719** [comment]
@Guoli-Lyu Glad to have you back. After version `0.5.0`, the `openapi.py` and `swagger.py` have been merged into one file. You can find exactly the same code snippet which you want to modify at current `swagger.py`.
https://github.com/huge-success/sanic-openapi/blob/f82af54c06ec1cb56d1e8ae361fa1ecc9656fe07/sanic_openapi/swagger.py#L78-L84
https://github.com/huge-success/sanic-openapi/blob/f82af54c06ec1cb56d1e8ae361fa1ecc9656fe07/sanic_openapi/swagger.py#L105-L106

So, what you have to do are:
1. Rebase your branch with current master.
2. Porting the changes you made from `openapi.py` to `swagger.py`.

If there is any problem, just let me know.

**Comment by Guoli-Lyu** [comment]
Ok, I have done rebasing and porting. @chenjr0719

**Comment by Guoli-Lyu** [comment]
What's the state of this PR? This PR have not merged yet.

**Comment by ahopkins** [comment]
@Guoli-Lyu Thanks for the nudge. I was waiting for the tests to complete and then forgot about it.

**Comment by inventionlabsSydney** [comment]
Just moved to Sanic and really hoping this PR can be sorted out :) 👍

**Comment by junhoyeo** [comment]
Hope it update soon

**Comment by ahopkins** [comment]
@junhoyeo what do you mean? it is merged.

**Comment by chenjr0719** [comment]
@ahopkins I think what @junhoyeo talking about is to publish a new release on PyPI.

**Pull request merged**